### PR TITLE
Add persistant configuration + Themes

### PR DESCRIPTION
**Pull request by sigmaSd**
From https://github.com/lemunozm/termchat/issues/37

### CODE REVIEW

**Comment by sigmaSd** [comment]
This is how the default configuration file looks now
```toml
discovery_addr = "238.255.0.1:5877"
tcp_server_port = 0
user_name = "mrcool"
terminal_bell = true

[theme]
message_colors = ["Blue", "Yellow", "Cyan", "Magenta"]
my_user_color = "Green"
date_color = "DarkGray"
system_info_color = ["Cyan", "LightCyan"]
system_warning_color = ["Yellow", "LightYellow"]
system_error_color = ["Red", "LightRed"]
chat_panel_color = "White"
progress_bar_color = "LightGreen"
command_color = "LightYellow"
input_panel_color = "White"
```

**Comment by sigmaSd** [comment]
The next thing should be making termchat accept predefined themes, instead of the need of changing the code, or changing each color in the config file.

```
$Config/termchat would look like
               |
               |____ config
               |_____themes
                          |________default
                          |________light
                          |________other_custom_themes ..
```

with config having a theme entry like `theme = dark/light/custom_name` which will make theme search for it it in the aforementioned path.

I prefer to leave that to another pr.

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            config.tcp_server_port = tcp_server_port.parse().unwrap();
+        }
+        if let Some(user_name) = matches.value_of("username") {
+            config.user_name = user_name.parse().unwrap();
+        }
+        if matches.value_of("quiet-mode").is_some() {
+            config.terminal_bell = false;
+        }
+        if let Some(theme) = matches.value_of("theme") {
+            let theme = theme.to_lowercase();
```
Comment (lemunozm) [inline src/config.rs]: I think that the validator ensures that it is already in lowercase

**Review by lemunozm — APPROVED** [review]
I love how it has turned out!

Thanks again @sigmaSd for your contribution.
Sorry for the delay to check this.

**Comment by sigmaSd** [comment]
Don't worry about the delay, I see you're busy with message-io hopefully that works out well!

File: src/config.rs (hunk 1 above)
Comment (sigmaSd) [reply src/config.rs]: You're right, fixed

File: README.md [hunk 2]
```diff
@@ -85,3 +108,6 @@ Termchat treats messages containings the following commands in a special way:
 
 ***A:*** Yeah! You can run `termchat` passing the flag `--quiet-mode` or simple `-q`.
 
+***Q:*** **I can't see anything on my light themed desktop!!!**
+
+***A:*** You can use `termchat --theme light` also you can custimize colors individually via the config file
```
Comment (lemunozm) [inline README.md]:
```suggestion
***A:*** You can use `termchat --theme light`, also you can customize colors individually via the config file.
```

**Review by lemunozm — CHANGES_REQUESTED** [review]
Thanks to add also the documentation. I forgot it completely!

**Review by lemunozm — APPROVED** [review]

**Comment by lemunozm** [comment]
Hi @sigmaSd, I saw that the send_file integration test, is broken after these changes, could you fix it whenever you can? Thanks!

**Pull request merged**